File: sword_for_offer/codes/cpp/sfo_03_find_duplicate_numbers_in_an_array_s1/sfo_03_find_duplicate_numbers_in_an_array_s1.cpp
/*
* File: sfo_03_find_duplicate_numbers_in_an_array_s1.cpp
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

#include "../include/include.hpp"

// ===== Solution Code =====
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
        unordered_map<int, bool> map;
        for(int num : nums) {
            if(map[num]) return num;
            map[num] = true;
        }
        return -1;
    }
};

int main() {
    // ======= Test Case =======
    vector<int> nums = { 2, 3, 1, 0, 2, 5, 3 };
    // ====== Driver Code ======
    Solution* slt = new Solution();
    int res = slt->findRepeatNumber(nums);
    cout << res << endl;
    
    return 0;
}

File: sword_for_offer/codes/java/sfo_03_find_duplicate_numbers_in_an_array_s1/sfo_03_find_duplicate_numbers_in_an_array_s1.java
/*
* File: sfo_03_find_duplicate_numbers_in_an_array_s1.java
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

package sfo_03_find_duplicate_numbers_in_an_array_s1;

import include.*;
import java.util.*;

// ===== Solution Code =====
class Solution {
    public int findRepeatNumber(int[] nums) {
        Set<Integer> dic = new HashSet<>();
        for(int num : nums) {
            if(dic.contains(num)) return num;
            dic.add(num);
        }
        return -1;
    }
}

public class sfo_03_find_duplicate_numbers_in_an_array_s1 {
    public static void main(String[] args) {
        // ======= Test Case =======
        int[] nums = { 2, 3, 1, 0, 2, 5, 3 };
        // ====== Driver Code ======
        Solution slt = new Solution();
        int res = slt.findRepeatNumber(nums);
        System.out.println(res);
    }
}
